chantal/build.py

```diff
@@ -46,13 +46,21 @@ def build_job(args):
         else:
             shallow = ""
 
-        run_command("git clone " + shallow +
-                    shlex.quote(args.clone_location) +
-                    " " + args.work_location, base_env)
+        if args.treeish:
+            run_command("git init " + args.work_location, base_env)
+        else:
+            run_command("git clone " + shallow +
+                        shlex.quote(args.clone_location) +
+                        " " + args.work_location, base_env)
 
     os.chdir(args.work_location)
 
     if args.treeish:
+        run_command("git remote add origin " +
+                    shlex.quote(args.clone_location), base_env)
```

Comment:
i think this will fail if we have cloned a repo, since then the remote named origin is already there.

Reply:
`clone` and `add origin` will run in different cases. `clone` will only exec if there's no treeish and if so, `add origin` will.